astropy/coordinates/transformations/graph.py

```diff
@@ -125,9 +125,9 @@ def invalidate_cache(self):
         are added or removed, but will need to be called manually if
         weights on transforms are modified inplace.
         """
-        del self._cached_names
```

Comment:
I was confused about it, but see that deleting does not work if the `cached_property` has not been executed and thus nothing is in `self.__dict__` yet. Bit of a footgun, that.